warp10/src/main/java/io/warp10/script/functions/TOGTS.java

```diff
@@ -16,98 +16,265 @@
 
 package io.warp10.script.functions;
 
-import java.math.BigDecimal;
-import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
-import java.util.Map;
-
-import org.apache.thrift.TDeserializer;
-import org.apache.thrift.TException;
-import org.apache.thrift.protocol.TCompactProtocol;
-
 import io.warp10.continuum.gts.GTSDecoder;
 import io.warp10.continuum.gts.GTSEncoder;
 import io.warp10.continuum.gts.GTSHelper;
 import io.warp10.continuum.gts.GTSWrapperHelper;
 import io.warp10.continuum.gts.GeoTimeSerie;
 import io.warp10.continuum.store.thrift.data.GTSWrapper;
+import io.warp10.continuum.store.thrift.data.Metadata;
 import io.warp10.crypto.OrderPreservingBase64;
 import io.warp10.script.NamedWarpScriptFunction;
 import io.warp10.script.WarpScriptException;
 import io.warp10.script.WarpScriptStack;
 import io.warp10.script.WarpScriptStackFunction;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TCompactProtocol;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
 
 /**
  * Converts an encoder into a map of gts, one per type
  */
 public class TOGTS extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
   public TOGTS(String name) {
     super(name);
-  }  
+  }
 
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
+
+    Map<String, String> typeMap = null;
     Object top = stack.pop();
-    
-    if (!(top instanceof String) && !(top instanceof byte[]) && !(top instanceof GTSEncoder)) {
-      throw new WarpScriptException(getName() + " operates on a string, byte array or encoder.");
+
+    if (top instanceof Map) {
+      typeMap = new LinkedHashMap<>();
+      // this is a map to specify type by name, it should contain valid types
+      for (Map.Entry<Object, Object> entry: ((Map<Object, Object>) top).entrySet()) {
+        if (!(entry.getKey() instanceof String)) {
+          throw new WarpScriptException(getName() + " type MAP input must contains encoder name as keys.");
+        }
+        if (entry.getValue() instanceof String) {
+          String t = (String) entry.getValue();
+          if (t.equals("LONG") || t.equals("DOUBLE") || t.equals("BOOLEAN") || t.equals("STRING")) {
```

Comment:
BINARY should be included as it will be the responsability of the caller to decode said strings using ISO8859-1 instead of UTF-8